activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -128,7 +128,14 @@ def foreign_key_options
       def columns
         result = [["#{name}_id", type, options]]
         if polymorphic
-          result.unshift(["#{name}_type", :string, polymorphic_options])
+          options = polymorphic_options
+          if index
+            indexable_string_limit = Base.connection.max_indexable_string_limit
```

Comment:
Thanks for your comment.

> using a smaller limit as a default will require communicating this to users, as the change may cause data loss if they have long class names.

It's reasonable but where should we communicate to users?



> Also, your patch breaks the existing behavior that allows passing a type-column-specific `limit` value:

Good catch. I didn't notice it. It will be fixed by the following change:

```ruby
options = {limit: indexable_string_limit}.merge(options)
```

If we choose current approach, I'll apply the change.